---
f_generate_search_path:
  ros__parameters:
    type: system
    parts:
      f_generate_search_path_node
    modes:
      __DEFAULT__:
        f_generate_search_path_node: active.__DEFAULT__
      fd_spiral_high:
        f_generate_search_path_node: active.fd_spiral_high
      fd_spiral_low:
        f_generate_search_path_node: active.fd_spiral_low
      fd_spiral_medium:
        f_generate_search_path_node: active.fd_spiral_medium
      fd_unground:
        f_generate_search_path_node: inactive

f_generate_search_path_node:
  ros__parameters:
    type: node
    modes:
      __DEFAULT__:
        ros__parameters:
          spiral_altitude: 1.0
      fd_spiral_high:
        ros__parameters:
          spiral_altitude: 3.0
      fd_spiral_medium:
        ros__parameters:
          spiral_altitude: 2.0
      fd_spiral_low:
        ros__parameters:
          spiral_altitude: 1.0

f_inspect_pipeline:
  ros__parameters:
    type: system
    parts:
      f_inspect_pipeline_node
    modes:
      __DEFAULT__:
        f_inspect_pipeline_node: inactive
      fd_inspect_pipeline:
        f_inspect_pipeline_node: active.__DEFAULT__
      fd_unground:
        f_inspect_pipeline_node: inactive

f_inspect_pipeline_node:
  ros__parameters:
    type: node
    modes:
      __DEFAULT__:
        ros__parameters:
          random_param: 2.0

f_control_motion:
  ros__parameters:
    type: system
    parts:
      f_control_motion_node
    modes:
      __DEFAULT__:
        f_control_motion_node: active.__DEFAULT__
      fd_failure_thruster_1:
        f_control_motion_node: active.fd_failure_thruster_1
      fd_failure_thruster_2:
        f_control_motion_node: active.fd_failure_thruster_2
      fd_failure_thruster_3:
        f_control_motion_node: active.fd_failure_thruster_3
      fd_failure_thruster_4:
        f_control_motion_node: active.fd_failure_thruster_4
      fd_failure_thruster_5:
        f_control_motion_node: active.fd_failure_thruster_5
      fd_failure_thruster_6:
        f_control_motion_node: active.fd_failure_thruster_6
      fd_unground:
        f_control_motion_node: inactive

f_control_motion_node:
  ros__parameters:
    type: node
    modes:
      __DEFAULT__:
        ros__parameters:
          thruster1: 1
          thruster2: 1
          thruster3: 1
          thruster4: 1
          thruster5: 1
          thruster6: 1
      fd_failure_thruster_1:
        ros__parameters:
          thruster1: 0
          thruster2: 1
          thruster3: 1
          thruster4: 1
          thruster5: 1
          thruster6: 1
      fd_failure_thruster_2:
        ros__parameters:
          thruster1: 1
          thruster2: 0
          thruster3: 1
          thruster4: 1
          thruster5: 1
          thruster6: 1
      fd_failure_thruster_3:
        ros__parameters:
          thruster1: 1
          thruster2: 1
          thruster3: 0
          thruster4: 1
          thruster5: 1
          thruster6: 1
      fd_failure_thruster_4:
        ros__parameters:
          thruster1: 1
          thruster2: 1
          thruster3: 1
          thruster4: 0
          thruster5: 1
          thruster6: 1
      fd_failure_thruster_5:
        ros__parameters:
          thruster1: 1
          thruster2: 1
          thruster3: 1
          thruster4: 1
          thruster5: 0
          thruster6: 1
      fd_failure_thruster_6:
        ros__parameters:
          thruster1: 1
          thruster2: 1
          thruster3: 1
          thruster4: 1
          thruster5: 1
          thruster6: 0
